pkgs/by-name/tr/tribler/package.nix

```diff
@@ -1,131 +1,144 @@
+      npm install
+      npm run build
+    '';
   };
-  libtorrent = (python3.pkgs.toPythonModule (libtorrent-rasterbar-1_2_x)).python;
 in
-stdenv.mkDerivation (finalAttrs: {
+python3.pkgs.buildPythonApplication {
+  inherit version src;
   pname = "tribler";
-  version = "7.14.0";
+  pyproject = true;
 
-  src = fetchurl {
-    url = "https://github.com/Tribler/tribler/releases/download/v${finalAttrs.version}/Tribler-${finalAttrs.version}.tar.xz";
-    hash = "sha256-fQJOs9P4y71De/+svmD7YZ4+tm/bC3rspm7SbOHlSR4=";
-  };
+  build-system = with python3.pkgs; [
+    setuptools
+    setuptools-scm
+  ];
 
-  nativeBuildInputs = [
-    python3.pkgs.wrapPython
-    makeWrapper
-    # we had a "copy" of this in tribler's makeWrapper
-    # but it went out of date and broke, so please just use it directly
-    qt5.wrapQtAppsHook
+  dependencies = with python3.pkgs; [
+    # requirements.txt
+    bitarray
+    configobj
+    lz4
+    libtorrent-rasterbar
+    pillow
+    pony
+    pystray
+    pyipv8
+    ipv8-rust-tunnels
   ];
 
-  buildInputs = [ python3.pkgs.python ];
-
-  pythonPath =
-    [ libtorrent ]
-    ++ (with python3.pkgs; [
-      # requirements-core.txt
-      aiohttp
-      aiohttp-apispec
-      anyio
-      chardet
-      configobj
-      cryptography
-      decorator
-      faker
-      libnacl
-      lz4
-      marshmallow
-      netifaces
-      networkx
-      pony
-      psutil
-      pyasn1
-      pydantic_1
-      pyopenssl
-      pyyaml
-      sentry-sdk
-      service-identity
-      yappi
-      yarl
-      bitarray
-      filelock
-      (pyipv8.overrideAttrs (p: rec {
-        version = "2.10.0";
-        src = fetchPypi {
-          inherit (p) pname;
-          inherit version;
-          hash = "sha256-yxiXBxBiPokequm+vjsHIoG9kQnRnbsOx3mYOd8nmiU=";
-        };
-      }))
-      file-read-backwards
-      brotli
-      human-readable
-      # requirements.txt
-      pillow
-      pyqt5
-      pyqt5-sip
-      pyqtgraph
-      pyqtwebengine
-    ]);
-
-  installPhase = ''
-    mkdir -pv $out
-    # Nasty hack; call wrapPythonPrograms to set program_PYTHONPATH.
-    wrapPythonPrograms
-    cp -prvd ./* $out/
-    makeWrapper ${python3.pkgs.python}/bin/python $out/bin/tribler \
-        --set _TRIBLERPATH "$out/src" \
-        --set PYTHONPATH $out/src/tribler-core:$out/src/tribler-common:$out/src/tribler-gui:$program_PYTHONPATH \
-        --set NO_AT_BRIDGE 1 \
-        --chdir "$out/src" \
-        --add-flags "-O $out/src/run_tribler.py"
-
-    mkdir -p $out/share/applications $out/share/icons
-    cp $out/build/debian/tribler/usr/share/applications/org.tribler.Tribler.desktop $out/share/applications/
-    cp $out/build/debian/tribler/usr/share/pixmaps/tribler_big.xpm $out/share/icons/tribler.xpm
-    mkdir -p $out/share/copyright/tribler
-    mv $out/LICENSE $out/share/copyright/tribler
-  '';
+  configurePhase = ''
+    ln -s ${./pyproject.toml} pyproject.toml
+    ln -s ${./__main__.py} src/tribler/__main__.py
```

Comment:
You can cp the filesinstead of linkin 

That way the tribler derivation wont have so many dependencies in its runtime closure